packages/desktop-client/src/components/filters/EditFilter.js

```diff
@@ -0,0 +1,358 @@
+  useEffect(() => {
+    // Flash the scrollbar
+    if (scrollableEl.current) {
+      let el = scrollableEl.current;
+      let top = el.scrollTop;
+      el.scrollTop = top + 1;
+      el.scrollTop = top;
+    }
+
+    // Run it here
+    async function run() {
+      if (adding) {
+        let filter;
+        if (location.state.inputConds) {
+          filter = {
+            conditions_op: 'and',
+            conditions: location.state.inputConds,
+          };
+        } else {
+          filter = {
+            conditions_op: 'and',
+            conditions: [{ op: 'is', field: 'payee', value: null, type: 'id' }],
+          };
+        }
+
+        dispatch({ type: 'set-filter', filter });
+      } else {
+        let filter = await loadFilter();
+        if (filter) {
+          dispatch({ type: 'set-filter', filter });
+        }
+      }
+    }
+    run();
+  }, []);
+
+  //Set Transactions preview table
+  useEffect(() => {
+    let unsubscribe;
+
+    send('make-filters-from-conditions', {
+      conditions: state.filter.conditions.map(unparse),
+    }).then(({ filters }) => {
+      if (filters.length > 0) {
+        const conditionsOpKey =
+          state.filter.conditionsOp === 'or' ? '$or' : '$and';
+        let live = liveQuery(
+          q('transactions')
+            .filter({ [conditionsOpKey]: filters })
+            .select('*'),
+          data => dispatch({ type: 'set-transactions', transactions: data }),
+        );
+        unsubscribe = live.unsubscribe;
+      }
+    });
+
+    return () => {
+      if (unsubscribe) {
+        unsubscribe();
+      }
+    };
+  }, [state.filter.conditions, state.filter.conditionsOp]);
+
+  let selectedInst = useSelected('transactions', state.transactions, []);
+
+  async function onSave() {
+    dispatch({ type: 'form-error', error: null });
+
+    let res = await sendCatch(adding ? 'filter/create' : 'filter/update', {
+      state: state.filter,
+      filters,
+    });
+
+    if (res.error) {
+      dispatch({
+        type: 'form-error',
+        error: res.error.message,
+        //'An error occurred while saving. Please contact help@actualbudget.com for support.',
+      });
+    } else {
+      //history.goBack();
+      history.push(location.state.locationPtr.pathname, {
+        callbackConditions: state.filter,
+      });
+    }
+  }
+
+  let editorStyle = {
+    backgroundColor: colors.n10,
+    borderRadius: 4,
+  };
+
+  return (
+    <Page title="Custom Filter" modalSize="medium">
+      <Stack direction="row" style={{ marginTop: 10 }}>
+        <FormField style={{ flex: 1 }}>
+          <FormLabel title="Filter Name" htmlFor="name-field" />
+          <GenericInput
+            field="string"
+            type="string"
+            value={state.filter.name}
+            onChange={e => {
+              dispatch({ type: 'set-field', field: 'name', value: e });
+            }}
+          />
+        </FormField>
+      </Stack>
+      <View style={{ flexShrink: 0 }}>
```

Comment:
yea, it crossed my mind as well. I noticed that schedules has a new format and that the rules editor could benefit from matching it. I think it may be best to deploy as a separate PR.